components/script/dom/bindings/buffer_source.rs

```diff
@@ -24,53 +26,15 @@ use crate::dom::globalscope::GlobalScope;
 use crate::script_runtime::JSContext;
 
 /// <https://webidl.spec.whatwg.org/#BufferSource>
-#[allow(dead_code)]
+/// The BufferSource typedef is used to represent objects that are either themselves an ArrayBuffer
+/// or ArrayBufferView which provide a view on to an ArrayBuffer.
 pub(crate) enum BufferSource {
-    Int8Array(Box<Heap<*mut JSObject>>),
-    Int16Array(Box<Heap<*mut JSObject>>),
-    Int32Array(Box<Heap<*mut JSObject>>),
-    Uint8Array(Box<Heap<*mut JSObject>>),
-    Uint16Array(Box<Heap<*mut JSObject>>),
-    Uint32Array(Box<Heap<*mut JSObject>>),
-    Uint8ClampedArray(Box<Heap<*mut JSObject>>),
-    BigInt64Array(Box<Heap<*mut JSObject>>),
-    BigUint64Array(Box<Heap<*mut JSObject>>),
-    Float32Array(Box<Heap<*mut JSObject>>),
-    Float64Array(Box<Heap<*mut JSObject>>),
-    DataView(Box<Heap<*mut JSObject>>),
+    ArrayBufferView(Box<Heap<*mut JSObject>>),
+    #[allow(dead_code)]
     ArrayBuffer(Box<Heap<*mut JSObject>>),
     Default(Box<Heap<*mut JSObject>>),
```

Comment:
I think it would be nice to document the variants with https://webidl.spec.whatwg.org/#ArrayBufferView and https://webidl.spec.whatwg.org/#idl-ArrayBuffer, and explain `Default`.